homeassistant/components/homekit/accessories.py

```diff
@@ -85,34 +85,57 @@ def set_accessory_info(acc, name, model, manufacturer=MANUFACTURER,
 class HomeAccessory(Accessory):
     """Adapter class for Accessory."""
 
-    # pylint: disable=no-member
+    category = 'OTHER'
 
-    def __init__(self, name=ACCESSORY_NAME, model=ACCESSORY_MODEL,
-                 category='OTHER', **kwargs):
+    def __init__(self, hass, name, entity_id, config, **kwargs):
         """Initialize a Accessory object."""
         super().__init__(name, **kwargs)
-        set_accessory_info(self, name, model)
-        self.category = getattr(Category, category, Category.OTHER)
+        set_accessory_info(self, name, model=entity_id)
+        self.category = getattr(Category, self.category, Category.OTHER)
+        self.entity_id = entity_id
+        self.hass = hass
+        self.init_setup(config)
+
+    def init_setup(self, config):
```

Comment:
This seems to be one step forward, one step back because of all the pylint disables....

If you still keep Class.category, you really have the same signature, so the super() call does not look half as bad!